### PR TITLE
[WIP]test c/common config pkg

### PR DESCRIPTION
**Pull request by QiWang19**
- Vendor in c/common config pkg for containers.conf https://github.com/containers/common/pull/20
- Set OCIRunime if runtime from c/common config is empty
- Vendor in [branch](https://github.com/containers/buildah/compare/v1.11.5...QiWang19:1.11.5_unshare_c/common) from build replace buildah/unshare with common/pkg/unshare to avoid unshare redefined both in buidlah&common under vendor folder error
- Remove libpod/pkg/config

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

**Comment by QiWang19** [comment]
@rhatdan @vrothberg @TomSweeneyRedHat PTAL

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/containernetworking/cni v0.7.2-0.20190904153231-83439463f784
 	github.com/containernetworking/plugins v0.8.2
 	github.com/containers/buildah v1.11.5
-	github.com/containers/conmon v2.0.2+incompatible // indirect
+	github.com/containers/common v0.0.2
```
Comment (TomSweeneyRedHat) [inline go.mod]: Is this version right?  I would have suspect v2.0.3?

Comment (QiWang19) [reply go.mod]: fixed. stays v2.0.2 the same as master.

Comment (TomSweeneyRedHat) [reply go.mod]: OK, but you've v0.0.2 here, not v2.0.2...

Comment (QiWang19) [reply go.mod]: conmon is v2.0.2 but common is v0.0.2

**Comment by vrothberg** [comment]
@baude @mheon PTAL

config management will now be in c/common

File: go.mod (hunk 1 above)
Comment (TomSweeneyRedHat) [reply go.mod]: TYVM @QiWang19 !  My eyes would never have spotted that with the particular font that GitHub uses by default for me.

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, but we may want to hold until after the next release.

**Comment by mheon** [comment]
Agree - I don't want this landing before a 1.7.x

File: libpod/common_test.go [hunk 2]
```diff
@@ -7,7 +7,8 @@ import (
 	"testing"
 	"time"
 
-	"github.com/containers/libpod/libpod/config"
+	"github.com/containers/common/pkg/config"
+
```
Comment (mheon) [inline libpod/common_test.go]: Why is this separate from other imports?

File: libpod/boltdb_state.go [hunk 3]
```diff
@@ -6,7 +6,8 @@ import (
 	"strings"
 	"sync"
 
-	"github.com/containers/libpod/libpod/config"
+	"github.com/containers/common/pkg/config"
+
```
Comment (mheon) [inline libpod/boltdb_state.go]: Same - why is this separate from other imports?

File: libpod/oci_conmon_linux.go [hunk 4]
```diff
@@ -68,10 +69,10 @@ func newConmonOCIRuntime(name string, paths []string, conmonPath string, runtime
 	runtime.name = name
 	runtime.conmonPath = conmonPath
 
-	runtime.conmonEnv = runtimeCfg.ConmonEnvVars
+	runtime.conmonEnv = runtimeCfg.Env
```
Comment (mheon) [inline libpod/oci_conmon_linux.go]: This doesn't seem right - this is specifically Conmon environment variables

File: libpod/options.go [hunk 5]
```diff
@@ -194,8 +195,8 @@ func WithConmonEnv(environment []string) RuntimeOption {
 			return define.ErrRuntimeFinalized
 		}
 
-		rt.config.ConmonEnvVars = make([]string, len(environment))
-		copy(rt.config.ConmonEnvVars, environment)
+		rt.config.Env = make([]string, len(environment))
```
Comment (mheon) [inline libpod/options.go]: Again, I don't think this is Env, this is specifically Conmon.

File: libpod/options.go [hunk 6]
```diff
@@ -345,7 +346,7 @@ func WithCNIConfigDir(dir string) RuntimeOption {
 			return define.ErrRuntimeFinalized
 		}
 
-		rt.config.CNIConfigDir = dir
+		rt.config.NetworkDir = dir
```
Comment (mheon) [inline libpod/options.go]: Really dislike the name here - this doesn't really tell me much. `NetworkConfigDir` maybe?

File: libpod/options.go [hunk 7]
```diff
@@ -358,7 +359,7 @@ func WithCNIPluginDir(dir string) RuntimeOption {
 			return define.ErrRuntimeFinalized
 		}
 
-		rt.config.CNIPluginDir = []string{dir}
+		rt.config.PluginDirs = []string{dir}
```
Comment (mheon) [inline libpod/options.go]: Again, don't like the name here - we have more than one type of plugin now. `NetworkPluginDir`? Actually, that probably doesn't work, there are non-CNI network plugins. Probably needs to be `CNIPluginDir` still.

File: libpod/runtime.go [hunk 8]
```diff
@@ -138,6 +140,15 @@ func newRuntimeFromConfig(ctx context.Context, userConfigPath string, options ..
 	if err != nil {
 		return nil, err
 	}
+
+	if conf.OCIRuntime == "" {
```
Comment (mheon) [inline libpod/runtime.go]: Woah. Why are we special-casing this? The config package should handle it.

**Comment by rhatdan** [comment]
This will not go in until 1.8 most likely.  It needs to get merged into Buildah fist to make sure Buildah and Podman do the correct thing on podman build.

Hopefully we can roll forward with this in Buildah soon.  Although I am not sure if everything works correctly without a config file, and that the config files merge properly yet.

We want the default to be everything commented out if possible, and then only have /usr/share/containers/containers.conf has specific settings for a distribution.

/usr/share/containers/containers.conf, /etc/containers/containers.conf and ~/.config/containers/containers.conf need to be able to be merged correctly for this to fully work.

Also containers.conf format has to be fully agreed upon before we release.

**Pull request closed without merging**